app/controllers/application_controller.rb

```diff
@@ -1,4 +1,13 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::Base
+  before_action :set_locale
+
+  def set_locale
+    I18n.locale = params[:locale] || I18n.default_locale
+  end
+
+  def default_url_options
+    { locale: I18n.locale }
+  end
```

Comment:
Looking at [the Rails Guide](https://guides.rubyonrails.org/i18n.html#translating-model-content) I don't think this is necessary given your changes to `routes.rb`, as it adds it as as HTTP `GET` param in addition to the other change's URL path.

Reply:
Removed